WcaOnRails/app/webpacker/components/CompetitionForm/NearbyComps.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable react/no-danger */
+import React, { useEffect, useState } from 'react';
+import { fetchJsonOrError } from '../../lib/requests/fetchWithAuthenticityToken';
+import { FieldWrapper } from './FormInputs';
+import { competitionNearbyJsonUrl } from '../../lib/requests/routes.js.erb';
+import I18n from '../../lib/i18n';
+import CompsTable from './CompsTable';
+
+export default function NearbyComps({
+  latData, longData,
+  startDateData, endDateData,
+}) {
+  const [nearby, setNearby] = useState();
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    if (!latData.value || !longData.value || !startDateData.value || !endDateData.value) return;
+    setLoading(true);
+    const params = new URLSearchParams();
+    params.append(`competition[${latData.attribute}]`, latData.value);
+    params.append(`competition[${longData.attribute}]`, longData.value);
+    params.append(`competition[${startDateData.attribute}]`, startDateData.value);
+    params.append(`competition[${endDateData.attribute}]`, endDateData.value);
+
+    fetchJsonOrError(`${competitionNearbyJsonUrl}?${params.toString()}`).then(({ data }) => {
+      setNearby(data);
+      setLoading(false);
+    });
+  }, [latData.value, longData.value, startDateData.value, endDateData.value]);
+
+  const label = I18n.t('competitions.adjacent_competitions.label', { days: 5, kms: 10 });
```

Comment:
Probably should extract the 5 and 10 as constants somewhere. I assume 5 days and 10 kms shows up elsewhere in the code, and ideally they would all have a common source of truth.